diff -c -P -r lpr-5.9.orig/COPYRIGHT lpr-5.9/COPYRIGHT
*** lpr-5.9.orig/COPYRIGHT	Mon Jan 15 17:38:51 1996
--- lpr-5.9/COPYRIGHT	Thu Feb  9 15:23:04 1995
***************
*** 0 ****
--- 1,3 ----
+ The lpr package is distributed under the BSD copyright which can be
+ found in this directory.
+ 
diff -c -P -r lpr-5.9.orig/Makefile lpr-5.9/Makefile
*** lpr-5.9.orig/Makefile	Mon Jan 15 17:38:51 1996
--- lpr-5.9/Makefile	Thu May 11 18:16:44 1995
***************
*** 0 ****
--- 1,27 ----
+ .EXPORT_ALL_VARIABLES:
+ 
+ CC = gcc
+ CFLAGS = -O2 -fomit-frame-pointer -m486 -include /usr/include/bsd/bsd.h \
+ 	-I/usr/include/bsd -I../common_source
+ LDFLAGS	= -N -s
+ LDLIBS = -lbsd -lcommon -L ../common_source
+ 
+ 
+ SUBDIRS	= common_source filters lpc lpd lpq lpr lprm lptest pac
+ 
+ all:
+ 	for i in $(SUBDIRS); do make -C $$i all; done
+ 
+ install:
+ 	install -o root -g root -m 755 -d ${DEBDIR}/var/spool
+ 	install -o root -g lp -m 2775 -d ${DEBDIR}/var/spool/lpd
+ 	install -o root -g lp -m 2775 -d ${DEBDIR}/var/spool/lpd/lp
+ 	install -o root -g lp -m 2775 -d ${DEBDIR}/var/spool/lpd/remote
+ 	install -o root -g root -m 0755 filter.pcl filter.ps ${DEBDIR}/etc
+ 	install -o root -g root -m 0644 etc.printcap ${DEBDIR}/etc/printcap
+ 	install -o root -g root -m 0644 printcap.5 ${DEBDIR}/usr/man/man5
+ 	for i in $(SUBDIRS); do make -C $$i install; done
+ 
+ clean:
+ 	for i in $(SUBDIRS); do make -C $$i clean; done
+ 
diff -c -P -r lpr-5.9.orig/common_source/Makefile lpr-5.9/common_source/Makefile
*** lpr-5.9.orig/common_source/Makefile	Mon Jan 15 17:38:51 1996
--- lpr-5.9/common_source/Makefile	Tue Mar  1 21:30:08 1994
***************
*** 0 ****
--- 1,11 ----
+ OBJS	= common.o displayq.o printcap.o rmjob.o startdaemon.o
+ 
+ all:	$(OBJS)
+ 	@ar rcv libcommon.a $(OBJS)
+ 	@ranlib libcommon.a
+ 
+ install: all
+ 
+ clean:
+ 	rm -f *.o *.a
+ 
diff -c -P -r lpr-5.9.orig/common_source/common.c lpr-5.9/common_source/common.c
*** lpr-5.9.orig/common_source/common.c	Mon Jan 15 17:38:51 1996
--- lpr-5.9/common_source/common.c	Mon Jan 15 17:46:31 1996
***************
*** 65,72 ****
  char	*PF;		/* name of vrast filter (per job) */
  char	*FF;		/* form feed string */
  char	*TR;		/* trailer string to be output when Q empties */
- short	SC;		/* suppress multiple copies */
- short	SF;		/* suppress FF on each print job */
  short	SH;		/* suppress header page */
  short	SB;		/* short banner instead of normal header */
  short	HL;		/* print header last */
--- 65,70 ----
***************
*** 78,85 ****
  short	BR;		/* baud rate if lp is a tty */
  int	FC;		/* flags to clear if lp is a tty */
  int	FS;		/* flags to set if lp is a tty */
- int	XC;		/* flags to clear for local mode */
- int	XS;		/* flags to set for local mode */
  short	RS;		/* restricted to those with local accounts */
  
  char	line[BUFSIZ];
--- 76,81 ----
***************
*** 182,188 ****
  getq(namelist)
  	struct queue *(*namelist[]);
  {
! 	register struct dirent *d;
  	register struct queue *q, **queue;
  	register int nitems;
  	struct stat stbuf;
--- 178,184 ----
  getq(namelist)
  	struct queue *(*namelist[]);
  {
! 	register struct direct *d;
  	register struct queue *q, **queue;
  	register int nitems;
  	struct stat stbuf;
***************
*** 192,198 ****
  
  	if ((dirp = opendir(SD)) == NULL)
  		return(-1);
! 	if (fstat(dirp->dd_fd, &stbuf) < 0)
  		goto errdone;
  
  	/*
--- 188,194 ----
  
  	if ((dirp = opendir(SD)) == NULL)
  		return(-1);
! 	if (fstat(dirfd(dirp), &stbuf) < 0)
  		goto errdone;
  
  	/*
Binary files lpr-5.9.orig/common_source/common.o and lpr-5.9/common_source/common.o differ
Binary files lpr-5.9.orig/common_source/displayq.o and lpr-5.9/common_source/displayq.o differ
Binary files lpr-5.9.orig/common_source/libcommon.a and lpr-5.9/common_source/libcommon.a differ
diff -c -P -r lpr-5.9.orig/common_source/lp.h lpr-5.9/common_source/lp.h
*** lpr-5.9.orig/common_source/lp.h	Mon Jan 15 17:38:51 1996
--- lpr-5.9/common_source/lp.h	Sat May 21 15:18:14 1994
***************
*** 41,47 ****
  #include <stdio.h>
  #include <sys/param.h>
  #include <sys/file.h>
! #include <dirent.h>
  #include <sys/stat.h>
  #include <sys/socket.h>
  #include <sys/un.h>
--- 41,47 ----
  #include <stdio.h>
  #include <sys/param.h>
  #include <sys/file.h>
! #include <sys/dir.h>
  #include <sys/stat.h>
  #include <sys/socket.h>
  #include <sys/un.h>
***************
*** 79,86 ****
  extern char	*CF;		/* name of cifplot filter (per job) */
  extern char	*FF;		/* form feed string */
  extern char	*TR;		/* trailer string to be output when Q empties */
- extern short	SC;		/* suppress multiple copies */
- extern short	SF;		/* suppress FF on each print job */
  extern short	SH;		/* suppress header page */
  extern short	SB;		/* short banner instead of normal header */
  extern short	HL;		/* print header last */
--- 79,84 ----
***************
*** 92,99 ****
  extern short	BR;		/* baud rate if lp is a tty */
  extern int	FC;		/* flags to clear if lp is a tty */
  extern int	FS;		/* flags to set if lp is a tty */
- extern int	XC;		/* flags to clear for local mode */
- extern int	XS;		/* flags to set for local mode */
  extern short	RS;		/* restricted to those with local accounts */
  
  extern char	line[BUFSIZ];
--- 90,95 ----
diff -c -P -r lpr-5.9.orig/common_source/lp.local.h lpr-5.9/common_source/lp.local.h
*** lpr-5.9.orig/common_source/lp.local.h	Mon Jan 15 17:38:51 1996
--- lpr-5.9/common_source/lp.local.h	Sat Jan 22 22:25:16 1994
***************
*** 47,56 ****
  #include <ar.h>
  
  #ifndef A_MAGIC1	/* must be a VM/UNIX system */
! #ifndef SARMAG
  #	undef ARMAG
  #	define ARMAG	0177545
- #endif
  #endif
  
  /*
--- 47,57 ----
  #include <ar.h>
  
  #ifndef A_MAGIC1	/* must be a VM/UNIX system */
! #	define A_MAGIC1 OMAGIC
! #	define A_MAGIC2 NMAGIC
! #	define A_MAGIC3 ZMAGIC
  #	undef ARMAG
  #	define ARMAG	0177545
  #endif
  
  /*
diff -c -P -r lpr-5.9.orig/common_source/pathnames.h lpr-5.9/common_source/pathnames.h
*** lpr-5.9.orig/common_source/pathnames.h	Mon Jan 15 17:38:51 1996
--- lpr-5.9/common_source/pathnames.h	Tue Sep  6 02:34:50 1994
***************
*** 37,49 ****
  #include <paths.h>
  
  #define	_PATH_DEFDEVLP		"/dev/lp"
! #define	_PATH_DEFSPOOL		"/var/spool/output/lpd"
  #define	_PATH_HOSTSEQUIV	"/etc/hosts.equiv"
  #define	_PATH_HOSTSLPD		"/etc/hosts.lpd"
! #define	_PATH_MASTERLOCK	"/var/spool/output/lpd.lock"
  #define	_PATH_PR		"/usr/bin/pr"
  #define	_PATH_PRINTCAP		"/etc/printcap"
! #define	_PATH_SOCKETNAME	"/var/run/printer"
  #define	_PATH_VFONT		"/usr/libdata/vfont/"
  #define	_PATH_VFONTB		"/usr/libdata/vfont/B"
  #define	_PATH_VFONTI		"/usr/libdata/vfont/I"
--- 37,49 ----
  #include <paths.h>
  
  #define	_PATH_DEFDEVLP		"/dev/lp"
! #define	_PATH_DEFSPOOL		"/var/spool/lpd"
  #define	_PATH_HOSTSEQUIV	"/etc/hosts.equiv"
  #define	_PATH_HOSTSLPD		"/etc/hosts.lpd"
! #define	_PATH_MASTERLOCK	"/var/run/lpd.pid"
  #define	_PATH_PR		"/usr/bin/pr"
  #define	_PATH_PRINTCAP		"/etc/printcap"
! #define	_PATH_SOCKETNAME	"/dev/printer"
  #define	_PATH_VFONT		"/usr/libdata/vfont/"
  #define	_PATH_VFONTB		"/usr/libdata/vfont/B"
  #define	_PATH_VFONTI		"/usr/libdata/vfont/I"
Binary files lpr-5.9.orig/common_source/printcap.o and lpr-5.9/common_source/printcap.o differ
diff -c -P -r lpr-5.9.orig/common_source/rmjob.c lpr-5.9/common_source/rmjob.c
*** lpr-5.9.orig/common_source/rmjob.c	Mon Jan 15 17:38:51 1996
--- lpr-5.9/common_source/rmjob.c	Thu Jun 30 19:19:21 1994
***************
*** 33,39 ****
  
  #ifndef lint
  /*static char sccsid[] = "from: @(#)rmjob.c	5.7 (Berkeley) 6/1/90";*/
! static char rcsid[] = "$Id: rmjob.c,v 1.3 1993/12/08 00:47:04 jtc Exp $";
  #endif /* not lint */
  
  /*
--- 33,39 ----
  
  #ifndef lint
  /*static char sccsid[] = "from: @(#)rmjob.c	5.7 (Berkeley) 6/1/90";*/
! static char rcsid[] = "$Id: rmjob.c,v 1.1 1994/05/23 09:04:58 rzsfl Exp rzsfl $";
  #endif /* not lint */
  
  /*
***************
*** 63,69 ****
  {
  	register int i, nitems;
  	int assasinated = 0;
! 	struct dirent **files;
  	char *cp;
  
  	if ((i = pgetent(line, printer)) < 0)
--- 63,69 ----
  {
  	register int i, nitems;
  	int assasinated = 0;
! 	struct direct **files;
  	char *cp;
  
  	if ((i = pgetent(line, printer)) < 0)
***************
*** 203,209 ****
  	char *file;
  {
  	register int *r, n;
! 	register char **u, *cp;
  	FILE *cfp;
  
  	/*
--- 203,209 ----
  	char *file;
  {
  	register int *r, n;
! 	char **u, *cp, p[64], h[64];
  	FILE *cfp;
  
  	/*
***************
*** 212,248 ****
  	if (strlen(file) < 7 || file[0] != 'c' || file[1] != 'f')
  		return(0);
  
- 	if (all && (from == host || !strcmp(from, file+6)))
- 		return(1);
- 
  	/*
  	 * get the owner's name from the control file.
  	 */
  	if ((cfp = fopen(file, "r")) == NULL)
  		return(0);
  	while (getline(cfp)) {
  		if (line[0] == 'P')
! 			break;
  	}
  	(void) fclose(cfp);
! 	if (line[0] != 'P')
! 		return(0);
  
  	if (users == 0 && requests == 0)
! 		return(!strcmp(file, current) && isowner(line+1, file));
  	/*
  	 * Check the request list
  	 */
  	for (n = 0, cp = file+3; isdigit(*cp); )
  		n = n * 10 + (*cp++ - '0');
  	for (r = requ; r < &requ[requests]; r++)
! 		if (*r == n && isowner(line+1, file))
  			return(1);
  	/*
  	 * Check to see if it's in the user list
  	 */
  	for (u = user; u < &user[users]; u++)
! 		if (!strcmp(*u, line+1) && isowner(line+1, file))
  			return(1);
  	return(0);
  }
--- 212,249 ----
  	if (strlen(file) < 7 || file[0] != 'c' || file[1] != 'f')
  		return(0);
  
  	/*
  	 * get the owner's name from the control file.
  	 */
+ 	p[0] = h[0] = '\0';
  	if ((cfp = fopen(file, "r")) == NULL)
  		return(0);
  	while (getline(cfp)) {
  		if (line[0] == 'P')
! 			strcpy(p, line+1);
! 		else if (line[0] == 'H')
! 			strcpy(h, line+1);
  	}
  	(void) fclose(cfp);
! 
! 	if (all && (from == host || !strcmp(from, h)))
! 		return(1);
  
  	if (users == 0 && requests == 0)
! 		return(!strcmp(file, current) && isowner(p, file, h));
  	/*
  	 * Check the request list
  	 */
  	for (n = 0, cp = file+3; isdigit(*cp); )
  		n = n * 10 + (*cp++ - '0');
  	for (r = requ; r < &requ[requests]; r++)
! 		if (*r == n && isowner(p, file, h))
  			return(1);
  	/*
  	 * Check to see if it's in the user list
  	 */
  	for (u = user; u < &user[users]; u++)
! 		if (!strcmp(*u, p) && isowner(p, file, h))
  			return(1);
  	return(0);
  }
***************
*** 253,264 ****
   * files sent from the remote machine to be removed.
   * Normal users can only remove the file from where it was sent.
   */
! isowner(owner, file)
! 	char *owner, *file;
  {
! 	if (!strcmp(person, root) && (from == host || !strcmp(from, file+6)))
  		return(1);
! 	if (!strcmp(person, owner) && !strcmp(from, file+6))
  		return(1);
  	if (from != host)
  		printf("%s: ", host);
--- 254,265 ----
   * files sent from the remote machine to be removed.
   * Normal users can only remove the file from where it was sent.
   */
! isowner(owner, file, h)
! 	char *owner, *file, *h;
  {
! 	if (!strcmp(person, root) && (from == host || !strcmp(from, h)))
  		return(1);
! 	if (!strcmp(person, owner) && !strcmp(from, h))
  		return(1);
  	if (from != host)
  		printf("%s: ", host);
Binary files lpr-5.9.orig/common_source/rmjob.o and lpr-5.9/common_source/rmjob.o differ
Binary files lpr-5.9.orig/common_source/startdaemon.o and lpr-5.9/common_source/startdaemon.o differ
diff -c -P -r lpr-5.9.orig/debian.Changelog lpr-5.9/debian.Changelog
*** lpr-5.9.orig/debian.Changelog	Mon Jan 15 17:38:51 1996
--- lpr-5.9/debian.Changelog	Sat Nov 25 16:05:26 1995
***************
*** 0 ****
--- 1,18 ----
+ Fri May 12 01:17:37 MET DST 1995  Peter Tobias  <tobias@et-inf.fho-emden.de>
+ 
+ 	lpr-5.9-6:
+ 
+ 	* moved filter.ps/filter.pcl files to /etc and added them to
+ 	  conffiles (Bug#858)
+ 
+ 	* Added a short notice about filter entries to /etc/printcap
+ 
+ 
+ 
+ Sat Nov 25 17:04:59 EST 1995   D.J. Gregor <dgregor@gregor.com>
+ 
+ 	lpr-5.9-7:
+ 
+ 	* removed if filter from etc.printcap
+ 
+ 	* changed maintainer
diff -c -P -r lpr-5.9.orig/debian.conffiles lpr-5.9/debian.conffiles
*** lpr-5.9.orig/debian.conffiles	Mon Jan 15 17:38:51 1996
--- lpr-5.9/debian.conffiles	Thu May 11 18:59:54 1995
***************
*** 0 ****
--- 1,5 ----
+ /etc/printcap
+ /etc/hosts.lpd
+ /etc/init.d/lpd
+ /etc/filter.pcl
+ /etc/filter.ps
diff -c -P -r lpr-5.9.orig/debian.control lpr-5.9/debian.control
*** lpr-5.9.orig/debian.control	Mon Jan 15 17:38:51 1996
--- lpr-5.9/debian.control	Mon Jan 15 17:39:35 1996
***************
*** 0 ****
--- 1,8 ----
+ PACKAGE: lpr
+ VERSION: 5.9
+ PACKAGE_REVISION: 11
+ MAINTAINER: D.J. Gregor <dgregor@coil.com>
+ DESCRIPTION: Berkeley lpr/lpd line printer spooling system
+  This is the standards UNIX printer spooler and associated utilties.
+  You can use this for local and remote printers.
+ DEPENDS: libc5
diff -c -P -r lpr-5.9.orig/debian.postinst lpr-5.9/debian.postinst
*** lpr-5.9.orig/debian.postinst	Mon Jan 15 17:38:51 1996
--- lpr-5.9/debian.postinst	Wed Mar 15 18:47:21 1995
***************
*** 0 ****
--- 1,14 ----
+ #!/bin/sh
+ # post install script for the Debian GNU/Linux lpr package
+ 
+ update-rc.d lpd defaults >/dev/null
+ 
+ /etc/init.d/lpd start
+ 
+ # Fix directory permissions that might be incorrect.
+ for i in /var/spool/lpd /var/spool/lpd/lp /var/spool/lpd/remote
+ do
+  if [ -d $i ]; then
+    chmod 2775 $i
+  fi
+ done
diff -c -P -r lpr-5.9.orig/debian.postrm lpr-5.9/debian.postrm
*** lpr-5.9.orig/debian.postrm	Mon Jan 15 17:38:51 1996
--- lpr-5.9/debian.postrm	Wed Mar 15 18:23:50 1995
***************
*** 0 ****
--- 1,10 ----
+ #!/bin/sh
+ # post remove script for the Debian GNU/Linux lpd package
+ 
+ set -e
+ 
+ if [ $1 = "purge" ]
+ then
+   update-rc.d lpd remove >/dev/null
+ fi
+  
diff -c -P -r lpr-5.9.orig/debian.preinst lpr-5.9/debian.preinst
*** lpr-5.9.orig/debian.preinst	Mon Jan 15 17:38:51 1996
--- lpr-5.9/debian.preinst	Wed Mar 15 19:46:07 1995
***************
*** 0 ****
--- 1,8 ----
+ #!/bin/sh
+ # pre install script for the Debian GNU/Linux lpr package
+ 
+ set -e
+ 
+ (kill -9 `cat /var/run/lpd.pid`) 2>/dev/null || exit 0
+ rm -f /var/run/lpd.pid
+ 
diff -c -P -r lpr-5.9.orig/debian.prerm lpr-5.9/debian.prerm
*** lpr-5.9.orig/debian.prerm	Mon Jan 15 17:38:51 1996
--- lpr-5.9/debian.prerm	Wed Mar 15 17:52:06 1995
***************
*** 0 ****
--- 1,7 ----
+ #!/bin/sh
+ # pre remove script for the Debian GNU/Linux lpd package
+ 
+ set -e
+ 
+ /etc/init.d/lpd stop
+  
diff -c -P -r lpr-5.9.orig/debian.rules lpr-5.9/debian.rules
*** lpr-5.9.orig/debian.rules	Mon Jan 15 17:38:51 1996
--- lpr-5.9/debian.rules	Sun Nov 26 12:15:00 1995
***************
*** 0 ****
--- 1,74 ----
+ #!/usr/bin/make -f
+ 
+ .PHONY: build
+ 
+ PKG = lpr-5.9
+ 
+ P_REV := $(shell grep "PACKAGE_REVISION:" debian.control | cut -d' ' -f 2)
+ 
+ 
+ # You have to use an absolute path for DEBDIR
+ DEBDIR = /tmp/debian-tmp
+ 
+ build:
+ 	make
+ 
+ install:
+ 	make install
+ 
+ clean:
+ 	build clean
+ 	rm -f debian-tmp
+ 
+ binary:
+ 	install -o root -g root -m 0755 -d ${DEBDIR}
+ 	install -o root -g root -m 0755 -d ${DEBDIR}/DEBIAN
+ 	install -o root -g root -m 0755 -d ${DEBDIR}/etc/init.d
+ 	install -o root -g root -m 0755 -d ${DEBDIR}/usr/sbin
+ 	install -o root -g root -m 0755 -d ${DEBDIR}/usr/bin
+ 	install -o root -g root -m 0755 -d ${DEBDIR}/usr/man/man1
+ 	install -o root -g root -m 0755 -d ${DEBDIR}/usr/man/man5
+ 	install -o root -g root -m 0755 -d ${DEBDIR}/usr/man/man8
+ 	install -o root -g root -m 0755 -d ${DEBDIR}/usr/doc/copyright
+ 	install -o root -g root -m 0755 -d ${DEBDIR}/var/log
+ 	install -o root -g root -m 0644 hosts.lpd ${DEBDIR}/etc
+ 	install -o root -g root -m 0755 rc.lpd ${DEBDIR}/etc/init.d/lpd
+ 	install -o root -g root -m 0644 COPYRIGHT ${DEBDIR}/usr/doc/copyright/lpr
+ 	install -o root -g root -m 0644 debian.control ${DEBDIR}/DEBIAN/control
+ 	install -o root -g root -m 0644 debian.conffiles ${DEBDIR}/DEBIAN/conffiles
+ 	install -o root -g root -m 0755 debian.preinst ${DEBDIR}/DEBIAN/preinst
+ 	install -o root -g root -m 0755 debian.postinst ${DEBDIR}/DEBIAN/postinst
+ 	install -o root -g root -m 0755 debian.prerm ${DEBDIR}/DEBIAN/prerm
+ 	install -o root -g root -m 0755 debian.postrm ${DEBDIR}/DEBIAN/postrm
+ 	touch ${DEBDIR}/var/log/lp-acct ${DEBDIR}/var/log/lp-errs
+ 	chmod 664 ${DEBDIR}/var/log/lp-acct ${DEBDIR}/var/log/lp-errs
+ 	make "DEBDIR=${DEBDIR}" install
+ 	dpkg --build ${DEBDIR}
+ 	mv ${DEBDIR}.deb ../${PKG}-${P_REV}.deb
+ 
+ source:
+ # Makes a source package.
+ 	chmod +x debian.rules
+ 	(cd .. && tar cvf ${PKG}-${P_REV}.tar ${PKG} && \
+ 	 gzip -9v ${PKG}-${P_REV}.tar) || exit 1
+ 
+ diff:	../${PKG}.orig
+ # Makes a diff package.
+ 	(cd .. && diff -c -P -r ${PKG}.orig ${PKG} \
+ 	 > ${PKG}-${P_REV}.diff; [ $$? = 1 ] && \
+ 	 gzip -9v ${PKG}-${P_REV}.diff) || \
+ 	 exit 1
+ 
+ md5:
+ 	(cd .. && \
+ 	md5sum ${PKG}-${P_REV}.deb >${PKG}-${P_REV}.md5 && \
+ 	md5sum ${PKG}-${P_REV}.diff.gz >>${PKG}-${P_REV}.md5 && \
+ 	md5sum ${PKG}-${P_REV}.tar.gz >>${PKG}-${P_REV}.md5)
+ 
+ # Make source, diff and binary packages
+ dist: source diff binary md5
+ 	cp debian.Changelog ../${PKG}-${P_REV}.Changelog
+ 
+ ../${PKG}.orig:
+ 	( cd .. && \
+ 	  tar xzf ${PKG}.orig.tar.gz )
diff -c -P -r lpr-5.9.orig/debian.todo lpr-5.9/debian.todo
*** lpr-5.9.orig/debian.todo	Mon Jan 15 17:38:51 1996
--- lpr-5.9/debian.todo	Sat Nov 25 16:09:28 1995
***************
*** 0 ****
--- 1,9 ----
+ This is the To Do list for the Debian ``lpr'' package:
+ 
+ 	- Add simple filter to distinguish between PostScript and text,
+ 	  like pscat on (yuk!) SCO UNIX.
+ 
+ 	- Fix Bug#1061 dealing with filters that differ from what they
+ 	  _should_ be.  This will take some work and some configuration
+ 	  script writing.
+ 
diff -c -P -r lpr-5.9.orig/etc.printcap lpr-5.9/etc.printcap
*** lpr-5.9.orig/etc.printcap	Mon Jan 15 17:38:51 1996
--- lpr-5.9/etc.printcap	Sat Nov 25 16:02:46 1995
***************
*** 1,31 ****
! #
! # Copyright (c) 1983 Regents of the University of California.
! # All rights reserved.
! #
! # Redistribution and use in source and binary forms are permitted
! # provided that this notice is preserved and that due credit is given
! # to the University of California at Berkeley. The name of the University
! # may not be used to endorse or promote products derived from this
! # software without specific prior written permission. This software
! # is provided ``as is'' without express or implied warranty.
! #
! #	@(#)etc.printcap	5.2 (Berkeley) 5/5/88
! #
! # DecWriter over a tty line.
! lp|ap|arpa|ucbarpa|LA-180 DecWriter III:\
! 	:br#1200:fs#06320:tr=\f:of=/usr/lib/lpf:lf=/usr/adm/lpd-errs:
! # typical remote printer entry
! ucbvax|vax|vx|ucbvax line printer:\
! 	:lp=:rm=ucbvax:sd=/usr/spool/vaxlpd:lf=/usr/adm/lpd-errs:
! varian|va|Benson Varian:\
! 	:lp=/dev/va0:sd=/usr/spool/vad:mx#2000:pl#58:px#2112:py#1700:tr=\f:\
! 	:of=/usr/lib/vpf:if=/usr/lib/vpf:tf=/usr/lib/rvcat:cf=/usr/lib/vdmp:\
! 	:gf=/usr/lib/vplotf:df=/usr/local/dvif:\
! 	:vf=/usr/lib/vpltdmp:lf=/usr/adm/lpd-errs:
! versatec|vp|Versatec plotter:\
! 	:lp=/dev/vp0:sd=/usr/spool/vpd:sb:sf:mx#0:pw#106:pl#86:px#7040:py#2400:\
! 	:of=/usr/lib/vpfW:if=/usr/lib/vpsf:tf=/usr/lib/vcat:cf=/usr/lib/vdmp:\
! 	:gf=/usr/lib/vplotf:vf=/usr/lib/vpltdmp:lf=/usr/adm/lpd-errs:\
! 	:tr=\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\
! \n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\
! \n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n:
--- 1,25 ----
! # /etc/printcap: printer capability database. See printcap(5).
! # You can use the filter entries df, tf, cf, gf etc. for
! # your own filters. See /etc/filter.ps, /etc/filter.pcl and
! # the printcap(5) manual page for further details.
! 
! lp|Generic dot-matrix printer entry:\
!         :lp=/dev/lp1:\
!         :sd=/var/spool/lpd/lp:\
!         :df=/etc/filter.ps:\
!         :tf=/etc/filter.pcl:\
!         :af=/var/log/lp-acct:\
!         :lf=/var/log/lp-errs:\
!         :pl#66:\
!         :pw#80:\
!         :pc#150:\
!         :mx#0:\
!         :sh:
! 
! # rlp|Remote printer entry:\
! #         :lp=:\
! #         :rm=remotehost:\
! #         :rp=remoteprinter:\
! #         :sd=/var/spool/lpd/remote:\
! #         :mx#0:\
! #         :sh:
diff -c -P -r lpr-5.9.orig/etc.printcap.bsd lpr-5.9/etc.printcap.bsd
*** lpr-5.9.orig/etc.printcap.bsd	Mon Jan 15 17:38:51 1996
--- lpr-5.9/etc.printcap.bsd	Wed Aug 10 17:36:10 1994
***************
*** 0 ****
--- 1,31 ----
+ #
+ # Copyright (c) 1983 Regents of the University of California.
+ # All rights reserved.
+ #
+ # Redistribution and use in source and binary forms are permitted
+ # provided that this notice is preserved and that due credit is given
+ # to the University of California at Berkeley. The name of the University
+ # may not be used to endorse or promote products derived from this
+ # software without specific prior written permission. This software
+ # is provided ``as is'' without express or implied warranty.
+ #
+ #	@(#)etc.printcap	5.2 (Berkeley) 5/5/88
+ #
+ # DecWriter over a tty line.
+ lp|ap|arpa|ucbarpa|LA-180 DecWriter III:\
+ 	:br#1200:fs#06320:tr=\f:of=/usr/lib/lpf:lf=/usr/adm/lpd-errs:
+ # typical remote printer entry
+ ucbvax|vax|vx|ucbvax line printer:\
+ 	:lp=:rm=ucbvax:sd=/usr/spool/vaxlpd:lf=/usr/adm/lpd-errs:
+ varian|va|Benson Varian:\
+ 	:lp=/dev/va0:sd=/usr/spool/vad:mx#2000:pl#58:px#2112:py#1700:tr=\f:\
+ 	:of=/usr/lib/vpf:if=/usr/lib/vpf:tf=/usr/lib/rvcat:cf=/usr/lib/vdmp:\
+ 	:gf=/usr/lib/vplotf:df=/usr/local/dvif:\
+ 	:vf=/usr/lib/vpltdmp:lf=/usr/adm/lpd-errs:
+ versatec|vp|Versatec plotter:\
+ 	:lp=/dev/vp0:sd=/usr/spool/vpd:sb:sf:mx#0:pw#106:pl#86:px#7040:py#2400:\
+ 	:of=/usr/lib/vpfW:if=/usr/lib/vpsf:tf=/usr/lib/vcat:cf=/usr/lib/vdmp:\
+ 	:gf=/usr/lib/vplotf:vf=/usr/lib/vpltdmp:lf=/usr/adm/lpd-errs:\
+ 	:tr=\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\
+ \n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\
+ \n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n:
diff -c -P -r lpr-5.9.orig/filter.pcl lpr-5.9/filter.pcl
*** lpr-5.9.orig/filter.pcl	Mon Jan 15 17:38:51 1996
--- lpr-5.9/filter.pcl	Wed Aug 24 14:02:12 1994
***************
*** 0 ****
--- 1,18 ----
+ #!/bin/sh
+ # This is a simple filter for PCL language printers, i.e., most
+ # HP DeskJet printers (see /etc/printcap).
+ 
+ # reset the printer to its defaults
+ echo -ne \\033\E
+ 
+ # to avoid stair-stepping, convert "LF" to "CR+LF"
+ echo -ne \\033\&k2G
+ 
+ # print text
+ cat
+ 
+ # form feed
+ echo -ne \\f
+ 
+ # reset the printer on exit
+ echo -ne \\033\E
diff -c -P -r lpr-5.9.orig/filter.ps lpr-5.9/filter.ps
*** lpr-5.9.orig/filter.ps	Mon Jan 15 17:38:51 1996
--- lpr-5.9/filter.ps	Wed Aug 24 03:25:21 1994
***************
*** 0 ****
--- 1,13 ----
+ #!/bin/sh
+ # This is a simple filter for printing postscript files, (see /etc/printcap).
+ 
+ # The 'epson' device is the generic device, for 9-pin and 24-pin printers.
+ /usr/bin/gs -q -dSAFER -sDEVICE=epson -dNOPAUSE -sOutputFile=- -
+ 
+ # 'eps9high' is a special mode for 9-pin printers where scan lines are
+ # interleaved in multiple passes to produce high vertical resolution at
+ # the expense of several passes of the print head.
+ #/usr/bin/gs -q -dSAFER -sDEVICE=eps9high -dNOPAUSE -sOutputFile=- -
+ 
+ # HP Deskjet 500 printer device
+ #/usr/bin/gs -q -dSAFER -sDEVICE=djet500 -dNOPAUSE -sOutputFile=- -
diff -c -P -r lpr-5.9.orig/filters/Makefile lpr-5.9/filters/Makefile
*** lpr-5.9.orig/filters/Makefile	Mon Jan 15 17:38:51 1996
--- lpr-5.9/filters/Makefile	Sun Sep  4 04:17:08 1994
***************
*** 0 ****
--- 1,10 ----
+ all:	lpf
+ 
+ lpf:	lpf.o
+ 
+ install: lpf
+ 	install -o root -g root -m 0755 lpf ${DEBDIR}/usr/bin
+ 
+ clean:
+ 	rm -f lpf *.o
+ 
Binary files lpr-5.9.orig/filters/lpf and lpr-5.9/filters/lpf differ
Binary files lpr-5.9.orig/filters/lpf.o and lpr-5.9/filters/lpf.o differ
diff -c -P -r lpr-5.9.orig/hosts.lpd lpr-5.9/hosts.lpd
*** lpr-5.9.orig/hosts.lpd	Mon Jan 15 17:38:51 1996
--- lpr-5.9/hosts.lpd	Wed Aug 17 14:36:22 1994
***************
*** 0 ****
--- 1,3 ----
+ # /etc/hosts.lpd: list of hosts that are allowed to use the printing
+ #		  services of this machine.  See lpd(8).
+ # $Id: hosts.lpd,v 2.0 1994/06/30 17:14:15 imurdock Exp $
diff -c -P -r lpr-5.9.orig/lpc/Makefile lpr-5.9/lpc/Makefile
*** lpr-5.9.orig/lpc/Makefile	Mon Jan 15 17:38:51 1996
--- lpr-5.9/lpc/Makefile	Tue Oct 25 18:44:53 1994
***************
*** 0 ****
--- 1,13 ----
+ LDFLAGS = -s
+ 
+ all:	lpc
+ 
+ lpc:	lpc.o cmds.o cmdtab.o
+ 
+ install: lpc
+ 	install -o root -g lp -m 2755 lpc ${DEBDIR}/usr/sbin
+ 	install -o root -g root -m 0644 lpc.8 ${DEBDIR}/usr/man/man8
+ 
+ clean:
+ 	rm -f lpc *.o
+ 
diff -c -P -r lpr-5.9.orig/lpc/cmds.c lpr-5.9/lpc/cmds.c
*** lpr-5.9.orig/lpc/cmds.c	Mon Jan 15 17:38:51 1996
--- lpr-5.9/lpc/cmds.c	Sat Jan 22 22:25:16 1994
***************
*** 47,53 ****
  /*
   * kill an existing daemon and disable printing.
   */
! abort(argc, argv)
  	char *argv[];
  {
  	register int c, status;
--- 47,53 ----
  /*
   * kill an existing daemon and disable printing.
   */
! xabort(argc, argv)
  	char *argv[];
  {
  	register int c, status;
***************
*** 208,215 ****
  	}
  }
  
! select(d)
! struct dirent *d;
  {
  	int c = d->d_name[0];
  
--- 208,215 ----
  	}
  }
  
! xselect(d)
! struct direct *d;
  {
  	int c = d->d_name[0];
  
***************
*** 223,229 ****
   * by `cf', `tf', or `df', then by the sequence letter A-Z, a-z.
   */
  sortq(d1, d2)
! struct dirent **d1, **d2;
  {
  	int c1, c2;
  
--- 223,229 ----
   * by `cf', `tf', or `df', then by the sequence letter A-Z, a-z.
   */
  sortq(d1, d2)
! struct direct **d1, **d2;
  {
  	int c1, c2;
  
***************
*** 247,253 ****
  {
  	register int i, n;
  	register char *cp, *cp1, *lp;
! 	struct dirent **queue;
  	int nitems;
  
  	bp = pbuf;
--- 247,253 ----
  {
  	register int i, n;
  	register char *cp, *cp1, *lp;
! 	struct direct **queue;
  	int nitems;
  
  	bp = pbuf;
***************
*** 259,265 ****
  		;
  	lp[-1] = '/';
  
! 	nitems = scandir(SD, &queue, select, sortq);
  	if (nitems < 0) {
  		printf("\tcannot examine spool directory\n");
  		return;
--- 259,265 ----
  		;
  	lp[-1] = '/';
  
! 	nitems = scandir(SD, &queue, xselect, sortq);
  	if (nitems < 0) {
  		printf("\tcannot examine spool directory\n");
  		return;
***************
*** 696,702 ****
  {
  	struct stat stbuf;
  	register int fd, i;
! 	register struct dirent *dp;
  	DIR *dirp;
  
  	bp = pbuf;
--- 696,702 ----
  {
  	struct stat stbuf;
  	register int fd, i;
! 	register struct direct *dp;
  	DIR *dirp;
  
  	bp = pbuf;
Binary files lpr-5.9.orig/lpc/cmds.o and lpr-5.9/lpc/cmds.o differ
diff -c -P -r lpr-5.9.orig/lpc/cmdtab.c lpr-5.9/lpc/cmdtab.c
*** lpr-5.9.orig/lpc/cmdtab.c	Mon Jan 15 17:38:51 1996
--- lpr-5.9/lpc/cmdtab.c	Sat Jan 22 22:25:16 1994
***************
*** 42,48 ****
  
  #include "lpc.h"
  
! int	abort(), clean(), enable(), disable(), down(), help();
  int	quit(), restart(), start(), status(), stop(), topq(), up();
  
  char	aborthelp[] =	"terminate a spooling daemon immediately and disable printing";
--- 42,48 ----
  
  #include "lpc.h"
  
! int	xabort(), clean(), enable(), disable(), down(), help();
  int	quit(), restart(), start(), status(), stop(), topq(), up();
  
  char	aborthelp[] =	"terminate a spooling daemon immediately and disable printing";
***************
*** 60,66 ****
  char	uphelp[] =	"enable everything and restart spooling daemon";
  
  struct cmd cmdtab[] = {
! 	{ "abort",	aborthelp,	abort,		1 },
  	{ "clean",	cleanhelp,	clean,		1 },
  	{ "enable",	enablehelp,	enable,		1 },
  	{ "exit",	quithelp,	quit,		0 },
--- 60,66 ----
  char	uphelp[] =	"enable everything and restart spooling daemon";
  
  struct cmd cmdtab[] = {
! 	{ "abort",	aborthelp,	xabort,		1 },
  	{ "clean",	cleanhelp,	clean,		1 },
  	{ "enable",	enablehelp,	enable,		1 },
  	{ "exit",	quithelp,	quit,		0 },
Binary files lpr-5.9.orig/lpc/cmdtab.o and lpr-5.9/lpc/cmdtab.o differ
Binary files lpr-5.9.orig/lpc/lpc and lpr-5.9/lpc/lpc differ
Binary files lpr-5.9.orig/lpc/lpc.o and lpr-5.9/lpc/lpc.o differ
diff -c -P -r lpr-5.9.orig/lpd/Makefile lpr-5.9/lpd/Makefile
*** lpr-5.9.orig/lpd/Makefile	Mon Jan 15 17:38:51 1996
--- lpr-5.9/lpd/Makefile	Wed Apr  5 04:21:46 1995
***************
*** 0 ****
--- 1,13 ----
+ LDFLAGS = -s
+ 
+ all:	lpd
+ 
+ lpd:	lpd.o lpdchar.o printjob.o recvjob.o
+ 
+ install: lpd
+ 	install -o root -g root -m 0755 lpd ${DEBDIR}/usr/sbin
+ 	install -o root -g root -m 0644 lpd.8 ${DEBDIR}/usr/man/man8
+ 
+ clean:
+ 	rm -f lpd *.o
+ 
Binary files lpr-5.9.orig/lpd/lpd and lpr-5.9/lpd/lpd differ
diff -c -P -r lpr-5.9.orig/lpd/lpd.c lpr-5.9/lpd/lpd.c
*** lpr-5.9.orig/lpd/lpd.c	Mon Jan 15 17:38:51 1996
--- lpr-5.9/lpd/lpd.c	Thu May 19 19:39:45 1994
***************
*** 134,140 ****
  		syslog(LOG_ERR, "%s: %m", _PATH_MASTERLOCK);
  		exit(1);
  	}
! 	signal(SIGCHLD, reapchild);
  	/*
  	 * Restart all the printers.
  	 */
--- 134,140 ----
  		syslog(LOG_ERR, "%s: %m", _PATH_MASTERLOCK);
  		exit(1);
  	}
! 	signal(SIGCHLD, SIG_IGN);
  	/*
  	 * Restart all the printers.
  	 */
***************
*** 178,183 ****
--- 178,184 ----
  		}
  		bzero(&sin, sizeof(sin));
  		sin.sin_family = AF_INET;
+ 		sin.sin_addr.s_addr = INADDR_ANY;
  		sin.sin_port = sp->s_port;
  		if (bind(finet, (struct sockaddr *)&sin, sizeof(sin)) < 0) {
  			syslog(LOG_ERR, "bind: %m");
Binary files lpr-5.9.orig/lpd/lpd.o and lpr-5.9/lpd/lpd.o differ
Binary files lpr-5.9.orig/lpd/lpdchar.o and lpr-5.9/lpd/lpdchar.o differ
diff -c -P -r lpr-5.9.orig/lpd/printjob.c lpr-5.9/lpd/printjob.c
*** lpr-5.9.orig/lpd/printjob.c	Mon Jan 15 17:38:52 1996
--- lpr-5.9/lpd/printjob.c	Sat May 21 15:22:44 1994
***************
*** 85,90 ****
--- 85,92 ----
  char	indent[10] = "-i0";	/* indentation size in characters */
  char	tempfile[] = "errsXXXXXX"; /* file name for filter output */
  
+ static short SF;		/* suppress FF on each print job */
+ 
  printjob()
  {
  	struct stat stb;
***************
*** 266,272 ****
  	 */
  	for (i = 0; i < 4; i++)
  		strcpy(fonts[i], ifonts[i]);
! 	strcpy(width+2, "0");
  	strcpy(indent+2, "0");
  
  	/*
--- 268,274 ----
  	 */
  	for (i = 0; i < 4; i++)
  		strcpy(fonts[i], ifonts[i]);
! 	sprintf(&width[2], "%d", PW);
  	strcpy(indent+2, "0");
  
  	/*
***************
*** 589,595 ****
  		while ((pid =
  		    wait3((int *)&status, WUNTRACED, 0)) > 0 && pid != ofilter)
  			;
! 		if (status.w_stopval != WSTOPPED) {
  			(void) close(fi);
  			syslog(LOG_WARNING, "%s: output filter died (%d)",
  				printer, status.w_retcode);
--- 591,597 ----
  		while ((pid =
  		    wait3((int *)&status, WUNTRACED, 0)) > 0 && pid != ofilter)
  			;
! 		if (!WIFSTOPPED(status)) {
  			(void) close(fi);
  			syslog(LOG_WARNING, "%s: output filter died (%d)",
  				printer, status.w_retcode);
***************
*** 1114,1123 ****
  		FC = 0;
  	if ((FS = pgetnum("fs")) < 0)
  		FS = 0;
- 	if ((XC = pgetnum("xc")) < 0)
- 		XC = 0;
- 	if ((XS = pgetnum("xs")) < 0)
- 		XS = 0;
  	tof = !pgetflag("fo");
  }
  
--- 1116,1121 ----
***************
*** 1230,1244 ****
   */
  setty()
  {
! 	struct sgttyb ttybuf;
  	register struct bauds *bp;
  
! 	if (ioctl(pfd, TIOCEXCL, (char *)0) < 0) {
! 		syslog(LOG_ERR, "%s: ioctl(TIOCEXCL): %m", printer);
! 		exit(1);
! 	}
! 	if (ioctl(pfd, TIOCGETP, (char *)&ttybuf) < 0) {
! 		syslog(LOG_ERR, "%s: ioctl(TIOCGETP): %m", printer);
  		exit(1);
  	}
  	if (BR > 0) {
--- 1228,1238 ----
   */
  setty()
  {
! 	struct termios ttybuf;
  	register struct bauds *bp;
  
! 	if (ioctl(pfd, TCGETS, (char *)&ttybuf) < 0) {
! 		syslog(LOG_ERR, "%s: ioctl(TCGETS): %m", printer);
  		exit(1);
  	}
  	if (BR > 0) {
***************
*** 1249,1273 ****
  			syslog(LOG_ERR, "%s: illegal baud rate %d", printer, BR);
  			exit(1);
  		}
! 		ttybuf.sg_ispeed = ttybuf.sg_ospeed = bp->speed;
  	}
! 	ttybuf.sg_flags &= ~FC;
! 	ttybuf.sg_flags |= FS;
! 	if (ioctl(pfd, TIOCSETP, (char *)&ttybuf) < 0) {
! 		syslog(LOG_ERR, "%s: ioctl(TIOCSETP): %m", printer);
  		exit(1);
- 	}
- 	if (XC) {
- 		if (ioctl(pfd, TIOCLBIC, &XC) < 0) {
- 			syslog(LOG_ERR, "%s: ioctl(TIOCLBIC): %m", printer);
- 			exit(1);
- 		}
- 	}
- 	if (XS) {
- 		if (ioctl(pfd, TIOCLBIS, &XS) < 0) {
- 			syslog(LOG_ERR, "%s: ioctl(TIOCLBIS): %m", printer);
- 			exit(1);
- 		}
  	}
  }
  
--- 1243,1255 ----
  			syslog(LOG_ERR, "%s: illegal baud rate %d", printer, BR);
  			exit(1);
  		}
! 		ttybuf.c_cflag = (ttybuf.c_cflag &= ~CBAUD) | bp->speed;
  	}
! 	ttybuf.c_cflag &= ~FC;          /* not quite right! */
! 	ttybuf.c_cflag |= FS;           /* not quite right! */
! 	if (ioctl(pfd, TCSETS, (char *)&ttybuf) < 0) {
! 		syslog(LOG_ERR, "%s: ioctl(TCSETS): %m", printer);
  		exit(1);
  	}
  }
  
Binary files lpr-5.9.orig/lpd/printjob.o and lpr-5.9/lpd/printjob.o differ
diff -c -P -r lpr-5.9.orig/lpd/recvjob.c lpr-5.9/lpd/recvjob.c
*** lpr-5.9.orig/lpd/recvjob.c	Mon Jan 15 17:38:52 1996
--- lpr-5.9/lpd/recvjob.c	Sat Jan 22 22:25:16 1994
***************
*** 245,250 ****
--- 245,253 ----
  chksize(size)
  	int size;
  {
+ #ifdef	__linux__
+ 	return(1);
+ #else
  	int spacefree;
  	struct statfs sfb;
  
***************
*** 257,262 ****
--- 260,266 ----
  	if (minfree + size > spacefree)
  		return(0);
  	return(1);
+ #endif
  }
  
  read_number(fn)
Binary files lpr-5.9.orig/lpd/recvjob.o and lpr-5.9/lpd/recvjob.o differ
diff -c -P -r lpr-5.9.orig/lpq/Makefile lpr-5.9/lpq/Makefile
*** lpr-5.9.orig/lpq/Makefile	Mon Jan 15 17:38:52 1996
--- lpr-5.9/lpq/Makefile	Tue Oct 25 18:45:09 1994
***************
*** 0 ****
--- 1,13 ----
+ LDFLAGS = -s
+ 
+ all:	lpq
+ 
+ lpq:	lpq.o
+ 
+ install: lpq
+ 	install -o root -g lp -m 6755 lpq ${DEBDIR}/usr/bin
+ 	install -o root -g root -m 0644 lpq.1 ${DEBDIR}/usr/man/man1
+ 
+ clean:
+ 	@rm -f lpq *.o
+ 
Binary files lpr-5.9.orig/lpq/lpq and lpr-5.9/lpq/lpq differ
Binary files lpr-5.9.orig/lpq/lpq.o and lpr-5.9/lpq/lpq.o differ
diff -c -P -r lpr-5.9.orig/lpr/Makefile lpr-5.9/lpr/Makefile
*** lpr-5.9.orig/lpr/Makefile	Mon Jan 15 17:38:52 1996
--- lpr-5.9/lpr/Makefile	Tue Oct 25 18:45:18 1994
***************
*** 0 ****
--- 1,13 ----
+ LDFLAGS = -s
+ 
+ all:	lpr
+ 
+ lpr:	lpr.o
+ 
+ install: lpr
+ 	install -o root -g lp -m 6755 lpr ${DEBDIR}/usr/bin
+ 	install -o root -g root -m 0644 lpr.1 ${DEBDIR}/usr/man/man1
+ 
+ clean:
+ 	rm -f lpr *.o
+ 
Binary files lpr-5.9.orig/lpr/lpr and lpr-5.9/lpr/lpr differ
diff -c -P -r lpr-5.9.orig/lpr/lpr.1 lpr-5.9/lpr/lpr.1
*** lpr-5.9.orig/lpr/lpr.1	Mon Jan 15 17:38:52 1996
--- lpr-5.9/lpr/lpr.1	Tue Aug 23 08:18:53 1994
***************
*** 204,223 ****
  Specifies an alternate default printer.
  .El
  .Sh FILES
! .Bl -tag -width /var/spool/output/*/tf* -compact
  .It Pa /etc/passwd
  Personal identification.
  .It Pa /etc/printcap
  Printer capabilities data base.
  .It Pa /usr/sbin/lpd*
  Line printer daemons.
! .It Pa /var/spool/output/*
  Directories used for spooling.
! .It Pa /var/spool/output/*/cf*
  Daemon control files.
! .It Pa /var/spool/output/*/df*
  Data files specified in "cf" files.
! .It Pa /var/spool/output/*/tf*
  Temporary copies of "cf" files.
  .El
  .Sh SEE ALSO
--- 204,223 ----
  Specifies an alternate default printer.
  .El
  .Sh FILES
! .Bl -tag -width /var/spool/lpd/*/tf* -compact
  .It Pa /etc/passwd
  Personal identification.
  .It Pa /etc/printcap
  Printer capabilities data base.
  .It Pa /usr/sbin/lpd*
  Line printer daemons.
! .It Pa /var/spool/lpd/*
  Directories used for spooling.
! .It Pa /var/spool/lpd/*/cf*
  Daemon control files.
! .It Pa /var/spool/lpd/*/df*
  Data files specified in "cf" files.
! .It Pa /var/spool/lpd/*/tf*
  Temporary copies of "cf" files.
  .El
  .Sh SEE ALSO
diff -c -P -r lpr-5.9.orig/lpr/lpr.c lpr-5.9/lpr/lpr.c
*** lpr-5.9.orig/lpr/lpr.c	Mon Jan 15 17:38:52 1996
--- lpr-5.9/lpr/lpr.c	Sat May 21 15:35:05 1994
***************
*** 57,62 ****
--- 57,65 ----
  #include <signal.h>
  #include <ctype.h>
  #include <syslog.h>
+ #include <unistd.h>
+ #include <stdlib.h>
+ #include <string.h>
  #include "lp.local.h"
  #include "pathnames.h"
  
***************
*** 94,100 ****
  char	*SD;			/* spool directory */
  char	*LO;			/* lock file name */
  char	*RG;			/* restrict group */
! short	SC;			/* suppress multiple copies */
  
  char	*getenv();
  char	*rindex();
--- 97,103 ----
  char	*SD;			/* spool directory */
  char	*LO;			/* lock file name */
  char	*RG;			/* restrict group */
! static short	SC;			/* suppress multiple copies */
  
  char	*getenv();
  char	*rindex();
***************
*** 568,573 ****
--- 571,577 ----
  		printf("%s: cannot open %s\n", name, file);
  		return(-1);
  	}
+ #ifndef	__linux__
  	if (read(fd, &execb, sizeof(execb)) == sizeof(execb)) {
  		if(!N_BADMAG(execb) || 
  #ifdef SARMAG
***************
*** 596,601 ****
--- 600,606 ----
  		}
  #endif
  	}
+ #endif
  	(void) close(fd);
  	if (rflag) {
  		if ((cp = rindex(file, '/')) == NULL) {
***************
*** 673,679 ****
  	register int c, len, fd, n;
  	register char *cp;
  	char buf[BUFSIZ];
- 	char *lmktemp();
  
  	(void) sprintf(buf, "%s/.seq", SD);
  	if ((fd = open(buf, O_RDWR|O_CREAT, 0661)) < 0) {
--- 678,683 ----
***************
*** 692,724 ****
  			n = n * 10 + (*cp++ - '0');
  		}
  	}
! 	len = strlen(SD) + strlen(host) + 8;
! 	tfname = lmktemp("tf", n, len);
! 	cfname = lmktemp("cf", n, len);
! 	dfname = lmktemp("df", n, len);
  	inchar = strlen(SD) + 3;
  	n = (n + 1) % 1000;
  	(void) lseek(fd, 0L, 0);
  	sprintf(buf, "%03d\n", n);
  	(void) write(fd, buf, strlen(buf));
  	(void) close(fd);	/* unlocks as well */
- }
- 
- /*
-  * Make a temp file name.
-  */
- char *
- lmktemp(id, num, len)
- 	char	*id;
- 	int	num, len;
- {
- 	register char *s;
- 	extern char *malloc();
- 
- 	if ((s = malloc(len)) == NULL)
- 		fatal("out of memory");
- 	(void) sprintf(s, "%s/%sA%03d%s", SD, id, num, host);
- 	return(s);
  }
  
  /*VARARGS1*/
--- 696,717 ----
  			n = n * 10 + (*cp++ - '0');
  		}
  	}
! 	len = strlen(SD) + 15;
! 	if ((cfname = malloc(len)) == NULL)
! 		fatal("out of memory");
! 	(void) sprintf(cfname, "%s/cfA%03dAXXXXXX", SD, n);
! 	if (mktemp(cfname) == NULL)
! 		fatal("couldn't create unique control file");
! 	if ((tfname = strdup(cfname)) == NULL || (dfname = strdup(cfname)) == NULL)
! 		fatal("out of memory");
  	inchar = strlen(SD) + 3;
+ 	tfname[inchar-2] = 't';
+ 	dfname[inchar-2] = 'd';
  	n = (n + 1) % 1000;
  	(void) lseek(fd, 0L, 0);
  	sprintf(buf, "%03d\n", n);
  	(void) write(fd, buf, strlen(buf));
  	(void) close(fd);	/* unlocks as well */
  }
  
  /*VARARGS1*/
Binary files lpr-5.9.orig/lpr/lpr.o and lpr-5.9/lpr/lpr.o differ
diff -c -P -r lpr-5.9.orig/lprm/Makefile lpr-5.9/lprm/Makefile
*** lpr-5.9.orig/lprm/Makefile	Mon Jan 15 17:38:52 1996
--- lpr-5.9/lprm/Makefile	Tue Oct 25 18:45:31 1994
***************
*** 0 ****
--- 1,13 ----
+ LDFLAGS = -s
+ 
+ all:	lprm
+ 
+ lprm:	lprm.o
+ 
+ install: lprm
+ 	install -o root -g lp -m 6755 lprm ${DEBDIR}/usr/bin
+ 	install -o root -g root -m 0644 lprm.1 ${DEBDIR}/usr/man/man1
+ 
+ clean:
+ 	rm -f lprm *.o
+ 
Binary files lpr-5.9.orig/lprm/lprm and lpr-5.9/lprm/lprm differ
Binary files lpr-5.9.orig/lprm/lprm.o and lpr-5.9/lprm/lprm.o differ
diff -c -P -r lpr-5.9.orig/lptest/Makefile lpr-5.9/lptest/Makefile
*** lpr-5.9.orig/lptest/Makefile	Mon Jan 15 17:38:52 1996
--- lpr-5.9/lptest/Makefile	Sun Sep  4 04:16:05 1994
***************
*** 0 ****
--- 1,11 ----
+ all:	lptest
+ 
+ lptest:	lptest.o
+ 
+ install: lptest
+ 	install -o root -g root -m 0755 lptest ${DEBDIR}/usr/bin
+ 	install -o root -g root -m 0644 lptest.1 ${DEBDIR}/usr/man/man1
+ 
+ clean:
+ 	rm -f lptest *.o
+ 
Binary files lpr-5.9.orig/lptest/lptest and lpr-5.9/lptest/lptest differ
Binary files lpr-5.9.orig/lptest/lptest.o and lpr-5.9/lptest/lptest.o differ
diff -c -P -r lpr-5.9.orig/pac/Makefile lpr-5.9/pac/Makefile
*** lpr-5.9.orig/pac/Makefile	Mon Jan 15 17:38:52 1996
--- lpr-5.9/pac/Makefile	Sun Sep  4 04:16:21 1994
***************
*** 0 ****
--- 1,11 ----
+ all:	pac
+ 
+ pac:	pac.o
+ 
+ install: pac
+ 	install -o root -g root -m 0755 pac ${DEBDIR}/usr/sbin
+ 	install -o root -g root -m 0644 pac.8 ${DEBDIR}/usr/man/man8
+ 
+ clean:
+ 	rm -f *.o pac
+ 
Binary files lpr-5.9.orig/pac/pac and lpr-5.9/pac/pac differ
diff -c -P -r lpr-5.9.orig/pac/pac.8 lpr-5.9/pac/pac.8
*** lpr-5.9.orig/pac/pac.8	Mon Jan 15 17:38:52 1996
--- lpr-5.9/pac/pac.8	Mon Mar 13 09:23:49 1995
***************
*** 87,96 ****
  usually, statistics are printed for every user who has used any paper.
  .El
  .Sh FILES
! .Bl -tag -width /var/account/?_sum -compact
! .It Pa /var/account/?acct
  raw accounting files
! .It Pa /var/account/?_sum
  summary accounting files
  .It Pa /etc/printcap
  printer capability data base
--- 87,96 ----
  usually, statistics are printed for every user who has used any paper.
  .El
  .Sh FILES
! .Bl -tag -width /var/log/?_sum -compact
! .It Pa /var/log/?acct
  raw accounting files
! .It Pa /var/log/?_sum
  summary accounting files
  .It Pa /etc/printcap
  printer capability data base
Binary files lpr-5.9.orig/pac/pac.o and lpr-5.9/pac/pac.o differ
diff -c -P -r lpr-5.9.orig/printcap.5 lpr-5.9/printcap.5
*** lpr-5.9.orig/printcap.5	Mon Jan 15 17:38:52 1996
--- lpr-5.9/printcap.5	Sat May 21 15:24:08 1994
***************
*** 0 ****
--- 1,285 ----
+ .\" Copyright (c) 1983, 1991 Regents of the University of California.
+ .\" All rights reserved.
+ .\"
+ .\" Redistribution and use in source and binary forms, with or without
+ .\" modification, are permitted provided that the following conditions
+ .\" are met:
+ .\" 1. Redistributions of source code must retain the above copyright
+ .\"    notice, this list of conditions and the following disclaimer.
+ .\" 2. Redistributions in binary form must reproduce the above copyright
+ .\"    notice, this list of conditions and the following disclaimer in the
+ .\"    documentation and/or other materials provided with the distribution.
+ .\" 3. All advertising materials mentioning features or use of this software
+ .\"    must display the following acknowledgement:
+ .\"	This product includes software developed by the University of
+ .\"	California, Berkeley and its contributors.
+ .\" 4. Neither the name of the University nor the names of its contributors
+ .\"    may be used to endorse or promote products derived from this software
+ .\"    without specific prior written permission.
+ .\"
+ .\" THIS SOFTWARE IS PROVIDED BY THE REGENTS AND CONTRIBUTORS ``AS IS'' AND
+ .\" ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ .\" IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ .\" ARE DISCLAIMED.  IN NO EVENT SHALL THE REGENTS OR CONTRIBUTORS BE LIABLE
+ .\" FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ .\" DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ .\" OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ .\" HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ .\" LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ .\" OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ .\" SUCH DAMAGE.
+ .\"
+ .\"     @(#)printcap.5	6.6 (Berkeley) 5/10/91
+ .\"
+ .Dd May 10, 1991
+ .Dt PRINTCAP 5
+ .Os BSD 4.2
+ .Sh NAME
+ .Nm printcap
+ .Nd printer capability data base
+ .Sh SYNOPSIS
+ .Nm printcap
+ .Sh DESCRIPTION
+ The
+ .Nm Printcap
+ function
+ is a simplified version of the
+ .Xr termcap 5
+ data base
+ used to describe line printers.  The spooling system accesses the
+ .Nm printcap
+ file every time it is used, allowing dynamic
+ addition and deletion of printers.  Each entry in the data base
+ is used to describe one printer.  This data base may not be
+ substituted for, as is possible for 
+ .Xr termcap ,
+ because it may allow accounting to be bypassed.
+ .Pp
+ The default printer is normally 
+ .Em lp ,
+ though the environment variable
+ .Ev PRINTER
+ may be used to override this.  Each spooling utility supports an option,
+ .Fl P Ar printer ,
+ to allow explicit naming of a destination printer.
+ .Pp
+ Refer to the
+ .%T "4.3 BSD Line Printer Spooler Manual"
+ for a complete discussion on how setup the database for a given printer.
+ .Sh CAPABILITIES
+ Refer to
+ .Xr termcap 5
+ for a description of the file layout.
+ .Bl -column Namexxx Typexx "/var/spool/lpdxxxxx"
+ .Sy Name	Type	Description
+ .It "af	str" Ta Dv NULL Ta No "name of accounting file"
+ .It "br	num	none	if lp is a tty, set the baud rate"
+ .Pf ( Xr ioctl 2
+ call)
+ .It "cf	str" Ta Dv NULL Ta No "cifplot data filter"
+ .It "df	str" Ta Dv NULL Ta No "tex data filter"
+ .Pf ( Tn DVI
+ format)
+ .It "fc	num	0	if lp is a tty, clear flag bits"
+ .Pq Pa sgtty.h
+ .It "ff	str" Ta So Li \ef Sc Ta No "string to send for a form feed"
+ .It "fo	bool	false	print a form feed when device is opened"
+ .It "fs	num	0	like `fc' but set bits"
+ .It "gf	str" Ta Dv NULL Ta No "graph data filter"
+ .Pf ( Xr plot 3
+ format
+ .It "hl	bool	false	print the burst header page last"
+ .It "ic	bool	false	driver supports (non standard) ioctl to indent printout"
+ .It "if	str" Ta Dv NULL Ta No "name of text filter which does accounting"
+ .It "lf	str" Ta Pa /dev/console Ta No "error logging file name"
+ .It "lo	str" Ta Pa lock Ta No "name of lock file"
+ .It "lp	str" Ta Pa /dev/lp Ta No "device name to open for output"
+ .It "mx	num	1000	maximum file size (in"
+ .Dv BUFSIZ
+ blocks), zero = unlimited
+ .It "nd	str" Ta Dv NULL Ta No "next directory for list of queues (unimplemented)"
+ .It "nf	str" Ta Dv NULL Ta No "ditroff data filter (device independent troff)"
+ .It "of	str" Ta Dv NULL Ta No "name of output filtering program"
+ .It "pc	num	200	price per foot or page in hundredths of cents"
+ .It "pl	num	66	page length (in lines)"
+ .It "pw	num	132	page width (in characters)"
+ .It "px	num	0	page width in pixels (horizontal)"
+ .It "py	num	0	page length in pixels (vertical)"
+ .It "rf	str" Ta Dv NULL Ta No "filter for printing"
+ .Tn FORTRAN
+ style text files
+ .It "rg	str" Ta Dv NULL Ta No "restricted group. Only members of group allowed access"
+ .It "rm	str" Ta Dv NULL Ta No "machine name for remote printer"
+ .It "rp	str	``lp''	remote printer name argument"
+ .It "rs	bool	false	restrict remote users to those with local accounts"
+ .It "rw	bool	false	open the printer device for reading and writing"
+ .It "sb	bool	false	short banner (one line only)"
+ .It "sc	bool	false	suppress multiple copies"
+ .It "sd	str" Ta Pa /var/spool/lpd Ta No "spool directory"
+ .It "sf	bool	false	suppress form feeds"
+ .It "sh	bool	false	suppress printing of burst page header"
+ .It "st	str" Ta Pa status Ta No "status file name"
+ .It "tf	str" Ta Dv NULL Ta No "troff data filter (cat phototypesetter)"
+ .It "tr	str" Ta Dv NULL Ta No "trailer string to print when queue empties"
+ .It "vf	str" Ta Dv NULL Ta No "raster image filter"
+ .El
+ .Pp
+ If the local line printer driver supports indentation, the daemon
+ must understand how to invoke it.
+ .Sh FILTERS
+ The
+ .Xr lpd 8
+ daemon creates a pipeline of
+ .Em filters
+ to process files for various printer types.
+ The filters selected depend on the flags passed to
+ .Xr lpr 1 .
+ The pipeline set up is:
+ .Bd -literal -offset indent
+ p	pr | if	regular text + pr(1)
+ none	if	regular text
+ c	cf	cifplot
+ d	df	DVI (tex)
+ g	gf	plot(3)
+ n	nf	ditroff
+ f	rf	Fortran
+ t	tf	troff
+ v	vf	raster image
+ .Ed
+ .Pp
+ The
+ .Sy if
+ filter is invoked with arguments:
+ .Bd -filled -offset indent
+ .Cm if
+ .Op Fl c
+ .Fl w Ns Ar width
+ .Fl l Ns Ar length
+ .Fl i Ns Ar indent
+ .Fl n Ar login
+ .Fl h Ar host acct-file
+ .Ed
+ .Pp
+ The
+ .Fl c
+ flag is passed only if the
+ .Fl l
+ flag (pass control characters literally)
+ is specified to
+ .Xr lpr .
+ The
+ .Ar Width
+ function
+ and
+ .Ar length
+ specify the page width and length
+ (from
+ .Cm pw
+ and
+ .Cm pl
+ respectively) in characters.
+ The
+ .Fl n
+ and
+ .Fl h
+ parameters specify the login name and host name of the owner
+ of the job respectively.
+ The
+ .Ar Acct-file
+ function
+ is passed from the
+ .Cm af
+ .Nm printcap
+ entry.
+ .Pp
+ If no
+ .Cm if
+ is specified,
+ .Cm of
+ is used instead,
+ with the distinction that
+ .Cm of
+ is opened only once,
+ while
+ .Cm if
+ is opened for every individual job.
+ Thus,
+ .Cm if
+ is better suited to performing accounting.
+ The
+ .Cm of
+ is only given the
+ .Ar width
+ and
+ .Ar length
+ flags.
+ .Pp
+ All other filters are called as:
+ .Bd -filled -offset indent
+ .Nm filter
+ .Fl x Ns Ar width
+ .Fl y Ns Ar length
+ .Fl n Ar login
+ .Fl h Ar host acct-file
+ .Ed
+ .Pp
+ where
+ .Ar width
+ and
+ .Ar length
+ are represented in pixels,
+ specified by the
+ .Cm px
+ and
+ .Cm py
+ entries respectively.
+ .Pp
+ All filters take
+ .Em stdin
+ as the file,
+ .Em stdout
+ as the printer,
+ may log either to
+ .Em stderr
+ or using
+ .Xr syslog 3 ,
+ and must not ignore
+ .Dv SIGINT . 
+ .Sh LOGGING
+ Error messages generated by the line printer programs themselves
+ (that is, the
+ .Xr lp Ns *
+ programs)
+ are logged by
+ .Xr syslog 3
+ using the
+ .Dv LPR
+ facility.
+ Messages printed on
+ .Em stderr
+ of one of the filters
+ are sent to the corresponding
+ .Cm lf
+ file.
+ The filters may, of course, use
+ .Xr syslog
+ themselves.
+ .Pp
+ Error messages sent to the console have a carriage return and a line
+ feed appended to them, rather than just a line feed.
+ .Sh SEE ALSO
+ .Xr termcap 5 ,
+ .Xr lpc 8 ,
+ .Xr lpd 8 ,
+ .Xr pac 8 ,
+ .Xr lpr 1 ,
+ .Xr lpq 1 ,
+ .Xr lprm 1
+ .Rs
+ .%T "4.3 BSD Line Printer Spooler Manual"
+ .Re
+ .Sh HISTORY
+ The
+ .Nm
+ file format appeared in
+ .Bx 4.2 ..
diff -c -P -r lpr-5.9.orig/rc.lpd lpr-5.9/rc.lpd
*** lpr-5.9.orig/rc.lpd	Mon Jan 15 17:38:52 1996
--- lpr-5.9/rc.lpd	Wed Mar 15 19:45:39 1995
***************
*** 0 ****
--- 1,21 ----
+ #!/bin/sh
+ #
+ # Start the printer daemon.
+ 
+ set -e
+ 
+ test -f /usr/sbin/lpd || exit 0
+ 
+ case "$1" in
+   start)
+ 	echo "Starting printer daemon" ; /usr/sbin/lpd
+ 	;;
+   stop)
+ 	start-stop-daemon --stop --quiet --oknodo --pidfile /var/run/lpd.pid --exec /usr/sbin/lpd
+ 	;;
+   *)
+         echo "Usage: /etc/init.d/lpd {start|stop}"
+         exit 1
+ esac
+ 
+ exit 0
